llama_index/llm_predictor/base.py

```diff
@@ -92,6 +98,16 @@ def predict(self, prompt: Prompt, **prompt_args: Any) -> str:
             # NOTE: this is an approximation, only for token counting
             formatted_prompt = messages_to_prompt(messages)
         else:
```

Comment:
This makes sense! My only concern is that chat-models are being ignored here.

We can at least support system prompts in the chat models super easily

Just have to prepend a system message to the list of messages

```
...
messages = prompt.format_messages(llm=self._llm, **prompt_args)
if system_prompt:
  messages = [ChatMessage(role=MessageRole.SYSTEM, content=system_prompt)] + messages
...
```